idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -536,127 +536,125 @@ def initialize(self, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
             solver (str): Solver to use for initialization
             optarg (dict): Solver arguments dictionary
         """
-        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
-        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
 
-        opt = SolverFactory(solver)
-        opt.options = optarg
-
-        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-            res = opt.solve(self, tee=slc.tee)
-        init_log.info_high(
-                "Initialization Step 1 {}.".format(idaeslog.condition(res))
-            )
+        flags = blk.control_volume.initialize(state_args=state_args,
+                                              hold_state=True)
 
-        for t in self.flowsheet().config.time:
-            for k, v in self.inlet.vars.items():
+        for t in blk.flowsheet().config.time:
```

Comment:
You do not need to do this here - the control volume initialization should have done this for you.